handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -1461,25 +1491,39 @@ public synchronized void setSSLParameters(SSLParameters sslParameters) {
         if (version >= 7) {
             endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
             algorithmConstraints = sslParameters.getAlgorithmConstraints();
-
-            if (version >= 8 && SNI_HOSTNAME_CLASS != null && clientMode && !isDestroyed()) {
-                assert GET_SERVER_NAMES_METHOD != null;
-                assert GET_ASCII_NAME_METHOD != null;
-                try {
-                    List<?> servernames = (List<?>) GET_SERVER_NAMES_METHOD.invoke(sslParameters);
-                    for (Object serverName : servernames) {
-                        if (SNI_HOSTNAME_CLASS.isInstance(serverName)) {
-                            SSL.setTlsExtHostName(ssl, (String) GET_ASCII_NAME_METHOD.invoke(serverName));
+            if (version >= 8) {
+                if (SNI_HOSTNAME_CLASS != null && clientMode && !isDestroyed()) {
+                    assert GET_SERVER_NAMES_METHOD != null;
+                    assert GET_ASCII_NAME_METHOD != null;
+                    try {
+                        List<?> servernames = (List<?>) GET_SERVER_NAMES_METHOD.invoke(sslParameters);
+                        for (Object serverName : servernames) {
+                            if (SNI_HOSTNAME_CLASS.isInstance(serverName)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 5 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)